etcdserver/api/v2http/client_auth.go

```diff
@@ -326,10 +326,10 @@ func (sh *authHandler) baseUsers(w http.ResponseWriter, r *http.Request) {
 		for _, roleName := range user.Roles {
 			var role auth.Role
 			role, err = sh.sec.GetRole(roleName)
-			if err != nil {
-				writeError(w, r, err)
-				return
+			if err != nil { // in case, role does not exist
+				plog.Warningf("GetRole %s error (%v)", roleName, err)
```

Comment:
i guess we should just remove the warning. this is nothing actually... etcd-users can assign a non-existing role to a user.